bodhi/server/scripts/skopeo_lite.py

```diff
@@ -0,0 +1,472 @@
+        if not os.path.exists(parent):
+            os.makedirs(parent)
+
+        return path
+
+    def get_blob(self, digest):
+        with open(self.get_blob_path(digest), 'rb') as f:
+            return f.read()
+
+    def has_blob(self, digest):
+        return os.path.exists(self.get_blob_path(digest))
+
+    def write_blob(self, digest, contents):
+        path = self.ensure_blob_path(digest)
+
+        with open(path, 'wb') as f:
+            f.write(contents)
+
+    def get_manifest(self, digest=None, media_type=None):
+        if digest is not None:
+            contents = self.get_blob(digest)
+        else:
+            manifest_path = os.path.join(self.directory, 'manifest.json')
+            if os.path.exists(manifest_path):
+                with open(manifest_path, 'rb') as f:
+                    contents = f.read()
+                parsed = json.loads(contents)
+                media_type = parsed.get('mediaType', MEDIA_TYPE_OCI)
+            else:
+                index_path = os.path.join(self.directory, 'index.json')
+                with open(index_path, 'rb') as f:
+                    contents = f.read()
+                parsed = json.loads(contents)
+                media_type = parsed.get('mediaType', MEDIA_TYPE_OCI_INDEX)
+
+        return ManifestInfo(contents, digest, media_type, len(contents))
+
+    def write_manifest(self, info, toplevel=False):
+        if not toplevel:
+            self.write_blob(info.digest, info.contents)
+        elif info.media_type in (MEDIA_TYPE_LIST_V2, MEDIA_TYPE_OCI_INDEX):
+            with open(os.path.join(self.directory, 'index.json'), 'wb') as f:
+                f.write(info.contents)
+        else:
+            with open(os.path.join(self.directory, 'manifest.json'), 'wb') as f:
+                f.write(info.contents)
+
+
+class RegistryEndpoint(object):
+    def __init__(self, spec):
+        self.session = spec.get_session()
+        self.registry = spec.registry
+        self.repo = spec.repo
+        self.tag = spec.tag
+
+    def start(self):
+        pass
+
+    def download_blob(self, digest, size, blob_path):
+        logger.info("%s: Downloading %s (size=%s)", self.registry, blob_path, size)
+
+        url = "/v2/{}/blobs/{}".format(self.repo, digest)
+        result = self.session.get(url, stream=True)
+        result.raise_for_status()
+
+        try:
+            with open(blob_path, 'wb') as f:
+                for block in result.iter_content(10 * 1024):
+                    f.write(block)
+        finally:
+            result.close()
+
+    def upload_blob(self, digest, size, blob_path):
+        logger.info("%s: Uploading %s (size=%s)", self.registry, blob_path, size)
+
+        url = "/v2/{}/blobs/uploads/".format(self.repo)
+        result = self.session.post(url, data='')
+        result.raise_for_status()
+
+        if result.status_code != requests.codes.ACCEPTED:
+            raise RuntimeError("Unexpected successful response %s", result.status_code)
```

Comment:
successful? Should it just be "Unexpected response"?

Reply:
if it was a failed response 4xx or 5xx then the raise_for_status() would have raised - so it's a "successful" response - but by the docker v2 api, a 202 ACCEPTED should be found here, not a 200 or other successful response. Alternate wording suggestions?

Reply:
I think the text is OK, but can you put your reply in here as a code comment for future readers?

Maybe "Unexpected successful response: %s (202 expected)".